### DISCUSSION TITLE
How to translate the base32hash js code into nushell?

### DISCUSSION THREAD

**Discussion by hustcer** [nushell/nushell#13847] [General]
I got this [JS code](https://github.com/pnpm/pnpm/blob/973e8845819abd45afd2bd7fb4f6948f1812463c/packages/crypto.base32-hash/src/index.ts#L6) to do base32hash from string content:

```javascript
import crypto from 'node:crypto'
import { base32 } from 'rfc4648'

export function createBase32Hash (str) {
  return base32.stringify(crypto.createHash('md5').update(str).digest()).replace(/(=+)$/, '').toLowerCase()
}

console.log(createBase32Hash('abcdefghijklmnopqrstuvwxyz'))
// The result should be: yp6nhv3bslsaa7p3jfwmuz7bhm
```
And try to rewrite it with Nushell but failed, any ideas? Thanks

**Comment by fdncred** [comment]
It looks to me like this needs to happen.
1. 'abcdefghijklmnopqrstuvwxyz' | hash md5
2. str replace -r '/(=+)$/' ''
3. str downcase
4. encode base32hex - but it doesn't work. not sure why

The md5 has here https://emn178.github.io/online-tools/base32_encode.html encodes as you would expect with hex input encoding. Ours works like this one https://cryptii.com/pipes/base32hex

**Reply by hustcer** [reply]
Yeah, I failed for the same reason. 😭

**Reply by weirdan** [reply]
```console
$ 'abcdefghijklmnopqrstuvwxyz' | hash md5 --binary | base32 | str replace -r '(=+)$' '' | str downcase
yp6nhv3bslsaa7p3jfwmuz7bhm
```

This uses an external `base32` binary. I don't seem to have `encode base32hex` - @fdncred, is it something new or maybe provided by a plugin?

**Reply by hustcer** [reply]
`encode base32hex` is newly added and will be available in v0.98. BTW: How do you install the `base32` binary?

**Reply by ghost** [reply]
`base32` is part of the gnu coreutils and should therefore be preinstalled on all mainstream linux distros (and in void, etc its prob in musl as well).

otherwise: busybox contains it as well and should be available for everything

**Comment by hustcer** [comment]
Thanks all, it's found that Nushell 0.97.2 can do it without any externals, That's awesome! :
```nu
'abcdefghijklmnopqrstuvwxyz' | hash md5 --binary | encode base32 --nopad | str downcase

```

**Reply by hustcer** [reply]
Special thanks to @KAAtheWiseGit 's PR: https://github.com/nushell/nushell/pull/13428 👍

**Reply by weirdan** [reply]
with `--nopad` you probably don't need that `str replace`.

**Reply by hustcer** [reply]
@weirdan Ah, that's great and much cleaner. 😎

**Reply by fdncred** [reply]
ah, the `hash md5 --binary` seems to have been the secret sauce. I didn't try that. Nice work!
